subcommand/get-consul-client-ca/command_test.go

```diff
@@ -0,0 +1,388 @@
+	require.True(t, roots.Roots[0].Active)
+	expectedCARoot := roots.Roots[0].RootCertPEM
+
+	// read the file contents
+	actualCARoot, err := ioutil.ReadFile(outputFile.Name())
+	require.NoError(t, err)
+	require.Equal(t, expectedCARoot, string(actualCARoot))
+}
+
+// Test that if the Consul server is not available at first,
+// we continue to poll it until it comes up.
+func TestRun_ConsulServerAvailableLater(t *testing.T) {
+	t.Parallel()
+	outputFile, err := ioutil.TempFile("", "ca")
+	require.NoError(t, err)
+
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI: ui,
+	}
+
+	randomPorts := freeport.MustTake(6)
+
+	// Start the command asynchronously
+	exitCode := -1
+	go func() {
+		exitCode = cmd.Run([]string{
+			"-server-addr", fmt.Sprintf("http://127.0.0.1:%d", randomPorts[1]),
+			"-output-file", outputFile.Name(),
+		})
+		require.Equal(t, 0, exitCode)
+	}()
+
+	// start the test server
+	time.Sleep(500 * time.Millisecond)
+	a, err := testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
+		c.Ports = &testutil.TestPortConfig{
+			DNS:     randomPorts[0],
+			HTTP:    randomPorts[1],
+			HTTPS:   randomPorts[2],
+			SerfLan: randomPorts[3],
+			SerfWan: randomPorts[4],
+			Server:  randomPorts[5],
+		}
+		c.Connect = map[string]interface{}{
+			"enabled": true,
+		}
+	})
+	require.NoError(t, err)
+	defer a.Stop()
+
+	// wait for command to exit
+	retry.Run(t, func(r *retry.R) {
+		require.Equal(r, 0, exitCode)
+	})
+
+	client, err := api.NewClient(&api.Config{
+		Address: a.HTTPAddr,
+	})
+	require.NoError(t, err)
+
+	// get the actual ca cert from consul
+	var expectedCARoot string
+	timer := &retry.Timer{Timeout: 500 * time.Millisecond, Wait: 100 * time.Millisecond}
+	retry.RunWith(timer, t, func(r *retry.R) {
+		roots, _, err := client.Agent().ConnectCARoots(nil)
+		require.NoError(r, err)
+		require.NotNil(r, roots)
+		require.NotNil(r, roots.Roots)
+		require.Len(r, roots.Roots, 1)
+		require.True(r, roots.Roots[0].Active)
+		expectedCARoot = roots.Roots[0].RootCertPEM
+	})
+
+	// check that the file contents match the actual CA
+	actualCARoot, err := ioutil.ReadFile(outputFile.Name())
+	require.NoError(t, err)
+	require.Equal(t, expectedCARoot, string(actualCARoot))
+}
+
+// Test that the command checks for the active root CA
+// and only writes the active one to the output file, ignoring
+// the inactive one.
+func TestRun_GetsOnlyActiveRoot(t *testing.T) {
+	t.Parallel()
+	outputFile, err := ioutil.TempFile("", "ca")
+	require.NoError(t, err)
+
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI: ui,
+	}
+
+	// start test server
+	a, err := testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
+		c.Connect = map[string]interface{}{
+			"enabled": true,
+		}
+	})
+	require.NoError(t, err)
+	defer a.Stop()
+
+	client, err := api.NewClient(&api.Config{
+		Address: a.HTTPAddr,
+	})
+	require.NoError(t, err)
+
+	// generate a new CA
+	ca, key := generateCA(t)
+
+	// set it as an active CA in Consul
```

Comment:
It would be helpful if the comment mentioned that doing this causes the current CA to become inactive, making there be two roots.